query/src/main/scala/filodb/query/ProtoConverters.scala

```diff
@@ -180,13 +180,28 @@ object ProtoConverters {
 
   implicit class PlannerParamsToProtoConverter(pp: PlannerParams) {
     def toProto: GrpcMultiPartitionQueryService.PlannerParams = {
+      val enforcedQuotaBuilder = GrpcMultiPartitionQueryService.IndividualQuota.newBuilder()
+      enforcedQuotaBuilder.setExecPlanSamples(pp.enforcedQuota.execPlanSamples)
+      enforcedQuotaBuilder.setExecPlanResultBytes(pp.enforcedQuota.execPlanResultBytes)
+      enforcedQuotaBuilder.setGroupByCardinality(pp.enforcedQuota.groupByCardinality)
+      enforcedQuotaBuilder.setJoinQueryCardinality(pp.enforcedQuota.joinQueryCardinality)
+      enforcedQuotaBuilder.setTimeSeriesSamplesScanBytes(pp.enforcedQuota.timeSeriesSamplesScanBytes)
+      val enforcedQuota = enforcedQuotaBuilder.build()
+      val warnQuotaBuilder = GrpcMultiPartitionQueryService.IndividualQuota.newBuilder()
+      warnQuotaBuilder.setExecPlanSamples(pp.warnQuota.execPlanSamples)
+      warnQuotaBuilder.setExecPlanResultBytes(pp.warnQuota.execPlanResultBytes)
+      warnQuotaBuilder.setGroupByCardinality(pp.warnQuota.groupByCardinality)
+      warnQuotaBuilder.setJoinQueryCardinality(pp.warnQuota.joinQueryCardinality)
+      warnQuotaBuilder.setTimeSeriesSamplesScanBytes(pp.warnQuota.timeSeriesSamplesScanBytes)
+      val warnQuota = warnQuotaBuilder.build()
       val builder = GrpcMultiPartitionQueryService.PlannerParams.newBuilder()
       builder.setApplicationId(pp.applicationId)
       builder.setQueryTimeoutMillis(pp.queryTimeoutMillis)
-      builder.setSampleLimit(pp.sampleLimit)
-      builder.setGroupByCardLimit(pp.groupByCardLimit)
-      builder.setJoinQueryCardLimit(pp.joinQueryCardLimit)
-      builder.setResultByteLimit(pp.resultByteLimit)
+      builder.setEnforcedQuota(enforcedQuota)
+      builder.setWarnQuota(warnQuota)
+      pp.queryOrigin.map(qo => builder.setQueryOrigin(qo))
```

Comment:
anti pattern for use of ``map``,  it should not cause side effects, either use ``isDefined`` or ``foreach``

Reply:
not sure I understood this comment. queryOrigin is an option. I set queryOrigin on the proto builder if and only if option is not None. Why side effects here?

Reply:
Your ``map`` is setting something in an object (side effect), by design this should simply transform and give a new option transforming the original one. As I mentioned, use ``foreach`` or ``isDefined`` to set this proto field.

Reply:
will do